admin/module/catalog-admin-module-sources/src/main/webapp/js/view/Source.view.js

```diff
@@ -78,6 +78,9 @@ function (ich,Marionette,_,$,Q,ModalSource,EmptyView,Service,Status,wreqr,Utils,
             }
             data.available = this.model.get('available');
             data.name = this.model.get('name');
+            if (!data.name && data.currentConfiguration && data.currentConfiguration.id) {
```

Comment:
The real problem is that this logic exists at all. Overriding serializeData is a code smell. The real problem is the sloppy binding between the model and the view. That seems outside the scope of this ticket, I wouldn't mind a ticket to clean up this whole module though.